flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/PushPartitionIntoTableSourceScanRule.scala

```diff
@@ -112,14 +125,28 @@ class PushPartitionIntoTableSourceScanRule extends RelOptRule(
     }
 
     val statistic = tableSourceTable.getStatistic
-    val newStatistic = if (remainingPartitions.size() == allPartitions.size()) {
-      // Keep all Statistics if no predicates can be pushed down
-      statistic
-    } else if (statistic == FlinkStatistic.UNKNOWN) {
-      statistic
-    } else {
-      // Remove tableStats after predicates pushed down
-      FlinkStatistic.builder().statistic(statistic).tableStats(null).build()
+    val newStatistic = {
+      val tableStats = catalogOption match {
+        case Some(catalog) =>
+          def mergePartitionStats(): TableStats = {
```

Comment:
why creating an inner function?

Reply:
Scala don't have code to break the loop.